cred/src/main/java/org/opencadc/cred/CredInitAction.java

```diff
@@ -116,62 +122,89 @@ public void doShutdown() {
+            }
+        }
+
+    }
     
     private void initConfig() {
-        this.credConfig = new CredConfig();
+        CredConfig credConfig = new CredConfig();
         PropertiesReader pr = new PropertiesReader(CONFIG_FILE);
         MultiValuedProperties mvp = pr.getAllProperties();
         if (mvp == null) {
             throw new RuntimeException("CONFIG: not found: " + CONFIG_FILE);
         }
-        
-        List<String> delegate = mvp.getProperty(DELEGATE_PROP);
-        if (delegate != null) {
-            for (String s : delegate) {
-                X500Principal p = new X500Principal(s);
-                credConfig.getDelegateUsers().add(p);
-            }
-        }
-        log.warn(DELEGATE_PROP + " found: " + credConfig.getDelegateUsers().size());
-        
-        List<String> proxy = mvp.getProperty(PROXY_PROP);
-        if (proxy != null) {
-            for (String s : proxy) {
-                X500Principal p = new X500Principal(s);
-                credConfig.getProxyUsers().add(p);
-            }
-        }
-        log.warn(PROXY_PROP + " found: " + credConfig.getProxyUsers().size());
-        
+
         String smax = mvp.getFirstPropertyValue(MAX_VALID_PROP);
         if (smax != null) {
             try {
                 float maxDaysValid = Float.parseFloat(smax);
                 if (maxDaysValid <= 0.0) {
                     throw new RuntimeException("CONFIG: invalid " + MAX_VALID_PROP + " = " + maxDaysValid + " -- must be positive");
                 }
-                credConfig.proxyMaxDaysValid = maxDaysValid;
+                credConfig.maxDaysValid = maxDaysValid;
             } catch (NumberFormatException ex) {
                 throw new RuntimeException("CONFIG: invalid " + MAX_VALID_PROP + " = " + smax, ex);
             }
         }
-        log.warn(MAX_VALID_PROP + " value: " + credConfig.proxyMaxDaysValid);
-        
+
+        log.debug(MAX_VALID_PROP + " value: " + credConfig.maxDaysValid);
+
+        if (SIGN_CERT_FILE.exists() && SIGN_CERT_FILE.canRead()) {
+            CheckCertificate checkCert = new CheckCertificate(SIGN_CERT_FILE);
+            try {
+                checkCert.check();
+                SSLUtil.readPemCertificateAndKey(SIGN_CERT_FILE);
+                credConfig.signingCert = SIGN_CERT_FILE.getAbsolutePath();
+            } catch (CheckException | CertificateException | InvalidKeySpecException | NoSuchAlgorithmException | IOException e) {
+                throw new RuntimeException("Invalid signing cert: " + SIGN_CERT_FILE.getPath(), e);
+            }
+        } else {
+            throw new RuntimeException("Signing cert not found or unreadable at: " + SIGN_CERT_FILE.getAbsolutePath());
+        }
+
+        log.debug("Signing cert: " + SIGN_CERT_FILE.getAbsolutePath());
+
+        for (String superuser : mvp.getProperty(SUPERUSER)) {
+            credConfig.superUsers.add(new X500Principal(superuser));
```

Comment:
I think to make this set work reliably you need to put these DNs into canonical form; iirc X500Principal allows both forward and reverse form and the `AuthenticationUtil.equals(Principal, Principal)` method suggests that we found comparisons a pain in the past (looking at it, that method is poorly written, but it does covert to canonical form).

That also means the callerDN that is checked for being in this set also needs to be in canonical form and I'd probably convert the userDN (in GetCertAction) to canonical form much earlier (to be safe if nothing else).

Reply:
I've tried it with different forms and it worked so I assumed that it was format was taken care of. However, to err on the conservative side, I've updated the code to explicitly use canonical forms to avoid any of these possible subtle problems.